### PR TITLE
More status improvements

### PR DESCRIPTION
**Pull request by russgold**
This change features more refactoring to simplify the code, and also aligns events with failure condition: a corresponding domain-failure event is automatically generated whenever a failure is added to the status.

### CODE REVIEW

**Comment by rjeberhard** [comment]
@russgold, the change looks good. I don't quite understand how knowing the `introspectionUid` of the last failed introspection helps the customer. What would the customer do with this?

**Comment by russgold** [comment]
> @russgold, the change looks good. I don't quite understand how knowing the `introspectionUid` of the last failed introspection helps the customer. What would the customer do with this?

The increment logic in `main` relies on the job UID to ensure that we don't increment it more than once for the job failure. Since my intent is to move that logic to be part of the addition of the condition, I figured that I would need the condition to have the UID. It is possible that I won't actually have it work that way, in which case all I will need is knowledge that the failure can be retried a finite number of times.

**Review by rjeberhard — APPROVED** [review]

File: kubernetes/crd/domain-crd.yaml [hunk 1]
```diff
@@ -11860,6 +11860,10 @@ spec:
                       description: Human-readable message indicating details about
                         last transition.
                       type: string
+                    introspectionUid:
```
[the remark this answers is not in the document]
Comment (russgold) [reply kubernetes/crd/domain-crd.yaml:11863]: I'll look into that for the next change, unless you feel it needs to be done as part of this one.

[the remark this answers is not in the document]
Comment (ankedia) [reply kubernetes/crd/domain-crd.yaml:11863]: ok, sure. Have you run the integration tests with these changes? If so, please update the PR with jenkins results link.

[the remark this answers is not in the document]
Comment (rjeberhard) [reply kubernetes/crd/domain-crd.yaml:11863]: I started a job: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9349/

**Pull request merged**